src/main/java/org/prebid/server/bidder/conversant/ConversantBidder.java

```diff
@@ -1,30 +1,239 @@
+                    + "Ignoring ImpID=%s", imp.getId()));
+        }
+    }
+
+    private static ExtImpConversant parseAndValidateImpExt(Imp imp) {
+        final ExtImpConversant extImpConversant;
+        try {
+            extImpConversant = Json.mapper.<ExtPrebid<?, ExtImpConversant>>convertValue(imp.getExt(),
+                    CONVERSANT_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+
+        if (StringUtils.isBlank(extImpConversant.getSiteId())) {
+            throw new PreBidException("Missing site id");
+        }
+
+        return extImpConversant;
+    }
+
+    private static Imp modifyImp(Imp imp, ExtImpConversant impExt) {
+        final BigDecimal extBidfloor = impExt.getBidfloor();
+        final String extTagId = impExt.getTagId();
+        final Integer extSecure = impExt.getSecure();
+        final boolean shouldChangeSecure = extSecure != null && (imp.getSecure() == null || imp.getSecure() == 0);
+        final Banner impBanner = imp.getBanner();
+        final Integer extPosition = impExt.getPosition();
+        final Video impVideo = imp.getVideo();
+
+        final Imp.ImpBuilder impBuilder = imp.toBuilder();
+
+        if (impBanner != null && extPosition != null) {
+            impBuilder.banner(impBanner.toBuilder().pos(extPosition).build());
+        }
+
+        return impBuilder
+                .displaymanager(DISPLAY_MANAGER)
+                .displaymanagerver(DISPLAY_MANAGER_VER)
+                .bidfloor(extBidfloor != null ? extBidfloor : imp.getBidfloor())
+                .tagid(extTagId != null ? extTagId : imp.getTagid())
+                .secure(shouldChangeSecure ? extSecure : imp.getSecure())
+                .video(impVideo != null ? modifyVideo(impVideo, impExt) : null)
+                .build();
+    }
+
+    private static Video modifyVideo(Video video, ExtImpConversant impExt) {
+        final List<String> extMimes = impExt.getMimes();
+        final Integer extMaxduration = impExt.getMaxduration();
+        final Integer extPosition = impExt.getPosition();
+        final List<Integer> extProtocols = impExt.getProtocols();
+        final List<Integer> extApi = impExt.getApi();
+        return video.toBuilder()
+                .mimes(extMimes != null ? extMimes : video.getMimes())
+                .maxduration(extMaxduration != null ? extMaxduration : video.getMaxduration())
+                .pos(makePosition(extPosition, video.getPos()))
+                .api(makeApi(extApi, video.getApi()))
+                .protocols(makeProtocols(extProtocols, video.getProtocols()))
+                .build();
+    }
+
+    private static Integer makePosition(Integer position, Integer videoPos) {
+        final Integer pos = videoPos != null && AD_POSITIONS.contains(videoPos) ? videoPos : null;
```

Comment:
i think it can be refactored a bit.

- add new method:
```
private static boolean isValidPosition(Integer position) {
  return position != null && AD_POSITIONS.contains(position);
}
```

- change `makePosition` to:
```
private static Integer makePosition(Integer position, Integer videoPos) {
  return isValidPosition(position) ? position : isValidPosition(videoPos) ? videoPos : null;
}
```
So, we'll check `videoPos` only if `position` fails.